### PR TITLE
[Homework #20] - Routing-redux by Julian Oviedo

### PR DESCRIPTION
**Pull request by JulianOviedo**
Hi Ilya ! Sorry for the delay, here is is my hw20... please, tell me if i need to improve something.

Thank you !

### CODE REVIEW

File: homeworks/julian-oviedo/20-routing-redux/todo-app/src/components/TodoList/TodoList.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+              </button>
+              <button className="list-group__button" onClick={toggleEdit}>
+                Cancel
+              </button>
+            </>
+          ))}
+    </div>
+  );
+};
+
```
[the remark this answers is not in the document]
Comment (JulianOviedo) [reply homeworks/julian-oviedo/20-routing-redux/todo-app/src/components/TodoList/TodoList.jsx:69]: I thought in that way was more clear to read but, you're right... i already remove it and fix the conditional logic

File: homeworks/julian-oviedo/20-routing-redux/todo-app/src/components/TodoList/TodoList.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      }),
+    );
+  }
+
+  function toggleEdit() {
+    setEditing(!editing);
+    const inputFocus = document.querySelector('#' + id);
+    inputFocus.focus();
+  }
+
```
[the remark this answers is not in the document]
Comment (JulianOviedo) [reply homeworks/julian-oviedo/20-routing-redux/todo-app/src/components/TodoList/TodoList.jsx:27]: Ok ok, done !

File: homeworks/julian-oviedo/20-routing-redux/todo-app/src/app/todoSlice.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    editTodo: (state, action) => {
+      return state.map((todo) => (todo.id === action.payload.id ? { ...todo, task: action.payload.newTodo } : todo));
+    },
+  },
+});
+
+export const { addTodo, deleteTodo, editTodo } = todoSlice.actions;
+
+export default todoSlice.reducer;
```
[the remark this answers is not in the document]
Comment (JulianOviedo) [reply homeworks/julian-oviedo/20-routing-redux/todo-app/src/app/todoSlice.js:25]: My bad. Done

**Review by ilya-ischenko — APPROVED** [review]
Well done!
Approved.

**Pull request merged**